ftdc/cmd/parser.go

```diff
@@ -0,0 +1,124 @@
+// main is a main package
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/ftdc"
+)
+
+// gnuplotWriter organizes all of the output for `gnuplot` to create a graph from FTDC
+// data. Notably:
+//
+//   - Each graph consists of all the readings for an individual metric. There is one file per metric
+//     and each file contains all of the (time, value) points to graph.
```

Comment:
The 1:1 graph per metric is a personal UX choice. Inspired entirely by the MDB ftdc/t2 experience.

> I think many graphs is probably fine as long as the x-axis for time remains consistent across a single fine so I can see relative timings of resource state changes.

You're right this is critical for usability. That invariant was held for every way I've shaped this data, but I never realized I depended on that being true. I'll try finding a spot to write that into the documentation somewhere.

I don't want too much exposition justifying my preferred UX (though I do think it's the only* right UX). Because I do think we did a good job of abstracting out a parser that anyone can use to present the data in any way they like. There's no reason we can't have multiple ways to render.

Reply:
Justifying this a bit further with the MDB's cloud monitoring graphs:
![image](https://github.com/user-attachments/assets/53dc616e-a19c-4d01-b849-e3343cee805b)

Reply:
Seems fair; thanks for the explanation here and offline. I agree that it's probably the best UX 🎉 .